### DISCUSSION TITLE
Welcome to auxilar_backend Discussions!

### DISCUSSION THREAD

**Discussion by vincent157664** [vincent157664/auxilar_backend#2] [Announcements]
Hello everyone. How are you?
Nice to meet you. I've opened this discussion to get your precious experiences related to deployment.

I've developed a MERN stack project and now I need to deploy it with GoDaddy domain. The project is already deployed on EC2 instance and now the task is to link GoDaddy domain and running project on EC2. 

I've connected frontend with domain on HTTP. But the backend isn't still connected and the server is running on HTTP. Please help me to solve this problem. Thanks.

**Comment by waytothevenus** [comment]
Hi Vincent. How are you? 
I am Andy, an experienced full-stack developer. I've faced the same problem when I deployed the first project. I've solved that problem using Nginx and certbot. You can install Nginx and certbot on your EC2 instance and config the nginx config file to support for frontend and backend. Please try and contact me if you face any problem.

**Reply by vincent157664** [reply]
Have you connected backend also?

**Reply by waytothevenus** [reply]
Of course.

**Comment by vincent157664** [comment]
Thanks. I will try again.

**Comment by vincent157664** [comment]
Could you please share config file?

**Comment by waytothevenus** [comment]
server {
    listen 80;
    server_name your_domain ;

    location / {
        return 301 https://$host$request_uri;
    }
}

# Redirect HTTP traffic to HTTPS for backend subdomain
server {
    listen 80;
    server_name sub_domain;

    location / {
        return 301 https://$host$request_uri;
    }
}

# Frontend server block - main domain
server {
    listen 443 ssl;
    server_name auxilar.org www.auxilar.org;

    # SSL configuration
    ssl_certificate /etc/letsencrypt/live/auxilar.org/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/auxilar.org/privkey.pem;
    include /etc/letsencrypt/options-ssl-nginx.conf;
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
 # Root directory for the frontend build files
    root /root/Auxilar-frontend/build;
    index index.html;

    location / {
        try_files $uri /index.html =404;
    }
}

# Backend server block - subdomain
server {
    listen 443 ssl;
    server_name backend.auxilar.org;

    # SSL configuration
    ssl_certificate /etc/letsencrypt/live/backend.auxilar.org/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/backend.auxilar.org/privkey.pem;
    include /etc/letsencrypt/options-ssl-nginx.conf;
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;

    location /api/ {
        proxy_pass http://localhost:3030;
        client_max_body_size 100M;
        proxy_http_version 1.1;
        proxy_set_header Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}

**Reply by waytothevenus** [reply]
Please locate this file in /etc/nginx/sites-available folder.

**Comment by vincent157664** [comment]
OK, thanks.

**Comment by vincent157664** [comment]
That works. Thanks for your effort.

**Comment by vincent157664** [comment]
Under the help of waytothevenus, I've deployed the MERN project successfully.
